src/lib/Dungeon.js

```diff
@@ -473,14 +555,14 @@ class AutomationDungeon
             }
         }
 
-        if (!this.__internal__isFirstMove)
+        if (!this.__internal__isFirstMove && !DungeonRunner.map.flash)
         {
             this.__internal__playerActionOccured = this.__internal__playerActionOccured
                                                 || !DungeonRunner.map.board()[currentFloor].every((row) => row.every((tile) => tile.isVisited));
         }
 
         // The boss was not found, reset the chest positions and move the player to the entrance, if not already there
-        if (!this.__internal__isCompleted && this.__internal__playerActionOccured)
+        if (!this.__internal__isCompleted && (this.__internal__playerActionOccured || DungeonRunner.map.flash))
```

Comment:
Why are we moving back to the starting tile in that case ?

Reply:
It was freezing if it hapened to cross a top corner without seeing the boss, but reading it again, this does not make sense. I will review if it is still needed, or if something else fixed that particular problem.